lib/Alchemy/Phrasea/Core/Provider/ORMServiceProvider.php

```diff
@@ -12,570 +12,207 @@
 namespace Alchemy\Phrasea\Core\Provider;
 
 use Alchemy\Phrasea\Application as PhraseaApplication;
+use Alchemy\Phrasea\Cache\Manager;
 use Alchemy\Phrasea\Core\Connection\ConnectionPoolManager;
-use Doctrine\Common\EventManager;
-use Doctrine\DBAL\Connection;
-use Doctrine\DBAL\DriverManager;
-use Doctrine\DBAL\Configuration;
 use Alchemy\Phrasea\Exception\InvalidArgumentException;
-use Alchemy\Phrasea\Model\NativeQueryProvider;
 use Doctrine\Common\Annotations\AnnotationReader;
+use Doctrine\Common\Annotations\AnnotationRegistry;
 use Doctrine\Common\Annotations\CachedReader;
-use Doctrine\Common\Cache\ArrayCache;
-use Doctrine\DBAL\Logging\DebugStack;
+use Doctrine\Common\Cache\Cache;
+use Doctrine\Common\EventManager;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DriverManager;
 use Doctrine\DBAL\Types\Type;
-use Doctrine\ORM\Configuration as ORMConfig;
 use Doctrine\ORM\EntityManager;
 use Doctrine\ORM\Mapping\Driver\AnnotationDriver;
+use Doctrine\ORM\Tools\Setup;
+use Gedmo\DoctrineExtensions;
 use Gedmo\Timestampable\TimestampableListener;
-use Monolog\Handler\RotatingFileHandler;
 use Silex\Application;
 use Silex\ServiceProviderInterface;
 
 class ORMServiceProvider implements ServiceProviderInterface
 {
-    public function register(Application $app)
-    {
-        // Provides DSN string using database information
-        $app['db.dsn'] = $app->protect(function(array $params) use ($app) {
-            $params = $app['db.info']($params);
-
-            switch ($params['driver']) {
-                case 'pdo_mysql':
-                    return sprintf('%s://%s:%s@%s:%s/%s',
-                        $params['driver'],
-                        $params['user'],
-                        $params['password'],
-                        $params['host'],
-                        $params['port'],
-                        $params['dbname']
-                    );
-                case 'pdo_sqlite':
-                    return sprintf('%s:%s',
-                        $params['driver'],
-                        $params['path']
-                    );
-                default:
-                    throw new \UnexpectedValueException(sprintf('Unknown driver "%s"', $params['driver']));
-            }
-        });
 
-        // Hash a DSN string
-        $app['hash.dsn'] = $app->protect(function($dsn) {
-            return md5($dsn);
-        });
+    private static $customTypes = [
+        'blob' => '\Alchemy\Phrasea\Model\Types\Blob',
```

Comment:
use `::class` constructs? Even `Types\Blob::class` to only import `Alchemy\Phrasea\Model\Types` namespace.